src/tx-builder/services/DydxStateChecker.ts

```diff
@@ -0,0 +1,29 @@
+import { DydxGovernor, DydxGovernor__factory } from '../../../types';
+import { Configuration } from '../types';
+
+export class DydxStateChecker {
+  readonly config: Configuration;
+
+  constructor(config: Configuration) {
+    this.config = config;
+  }
+
+  public async getGovernanceVoters(
+    governorContractAddress: string,
+    maxBlock: number,
+  ): Promise<Set<string>> {
+    const { provider }: Configuration = this.config;
+
+    const governor: DydxGovernor = await DydxGovernor__factory.connect(
+      governorContractAddress,
+      provider,
+    );
+
+    const filter = governor.filters.VoteEmitted(null, null, null, null);
+    const events = await governor.queryFilter(filter);
```

Comment:
nit: Can you use `maxBlock` within the call to `queryFilter` instead of filtering the events manually below? Docs here: https://docs.ethers.io/v5/api/contract/contract/#Contract-queryFilter

For the `fromBlock`, the deployment block of the `DydxGovernor` contract could work fine (or the block where the first proposal was created), however IMO taking `fromBlock` as an additional parameter makes the most sense. Up to you though!